scripts/release/homebrew/docker/formula_generate.py

```diff
@@ -85,6 +85,13 @@ def collect_resources() -> str:
 
 def collect_resources_dict() -> dict:
     nodes = make_graph('azure-cli')
+
+    # Homebrew does not install pip and setuptools after Python 3.12
+    # see https://github.com/Azure/azure-cli/pull/29887
+    extra_dependencies = ['pip', 'setuptools']
+    for dependency in extra_dependencies:
+        nodes[dependency] = research_package(dependency)
```

Comment:
`make_graph` explicitly exclude `pip` and `setuptools`: https://github.com/tdsmith/homebrew-pypi-poet/blob/fdafc615bcd28f29bcbe90789f07cc26f97c3bbc/poet/poet.py#L131

Adding `pip` and `setuptools` in CLI's dependency does not work as brew's `pip` uses `--no-deps` option.  https://github.com/Homebrew/homebrew-core/pull/194884